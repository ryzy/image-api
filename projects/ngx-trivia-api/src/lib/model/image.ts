export interface ImageLinks {
  /**
   * HTML link to the image (browse'able from browser)
   */
  html: string;

  large: string;
  regular: string;
  small: string;
  thumb: string;
}

/**
 * Object present when license is different than PublicDomain
 */
export interface ImageAuthor {
  name?: string;
  url?: string;
  avatar?: string;
  twitter?: string;
}

export interface Image {
  id: string;
  caption: string;
  source: 'Google' | 'Unsplash';
  license: 'cc_publicdomain' | 'cc_attribute';
  location?: string;
  links: ImageLinks;
  author?: ImageAuthor;
}
