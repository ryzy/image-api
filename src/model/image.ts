export interface ImageLinks {
  html: string;
  full: string;
  raw: string;
  regular: string;
  small: string;
  thumb: string;
}
export interface ImageAuthor {
  source: 'Unsplash';
  name: string;
  url: string;
  avatar: string;
  twitter: string;
}

export interface Image {
  id: string;
  caption: string;
  location: string;
  links: ImageLinks;
  author: ImageAuthor;
}

export function unsplashToImage(data: any = {}): Image {
  return {
    id: data.id,
    caption: data.description,
    location: data.location && data.location.title,
    links: {
      html: data.links && data.links.html,
      ...data.urls,
    },
    author: {
      source: 'Unsplash',
      name: data.user && data.user.name,
      url: data.user && data.user.links && data.user.links.html,
      avatar:
        data.user && data.user.profile_image && data.user.profile_image.large,
      twitter: data.user && data.user.twitter_username,
    },
  };
}
