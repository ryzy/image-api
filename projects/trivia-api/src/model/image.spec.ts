import { unsplashToImage } from './image';
import { mockImage1 } from '../../../../test/fixtures/image';
import { mockUnsplashImage } from '../../../../test/fixtures/unsplash-image';

describe('Image model', () => {
  test('#unsplashToImage', () => {
    expect(unsplashToImage()).toMatchSnapshot();
    expect(unsplashToImage(mockUnsplashImage)).toEqual(mockImage1);
  });
});
