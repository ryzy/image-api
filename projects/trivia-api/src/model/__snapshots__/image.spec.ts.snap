// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image model #unsplashToImage 1`] = `
Object {
  "author": Object {
    "avatar": undefined,
    "name": undefined,
    "source": "Unsplash",
    "twitter": undefined,
    "url": undefined,
  },
  "caption": undefined,
  "id": undefined,
  "links": Object {
    "html": undefined,
  },
  "location": undefined,
}
`;
